packages/manager/src/features/Account/SwitchAccountDrawer.tsx

```diff
@@ -1,104 +1,207 @@
+  const handleClose = React.useCallback(() => {
     onClose();
-  };
-
-  const { data: profile } = useProfile();
-  const { data: user } = useAccountUser(profile?.username ?? '');
-
-  // From proxy accounts, make a request on behalf of the parent account to fetch child accounts.
-  const headers =
-    flags.parentChildAccountAccess && user?.user_type === 'proxy'
-      ? new AxiosHeaders({ Authorization: authentication.token.get() }) // TODO: Parent/Child - M3-7430: replace this token with the parent token in local storage.
-      : undefined;
-  const { data: childAccounts, error, isLoading } = useChildAccounts({
-    headers,
-  });
-
-  const renderChildAccounts = React.useCallback(() => {
-    if (isLoading) {
-      return <CircleProgress mini />;
-    }
+  }, [onClose]);
 
-    if (childAccounts?.results === 0) {
-      return <Notice variant="info">There are no child accounts.</Notice>;
-    }
+  /**
+   * Headers are required for proxy users when obtaining a proxy token.
+   * For 'proxy' userType, use the stored parent token in the request.
+   */
+  const getProxyToken = React.useCallback(
+    async ({
+      euuid,
+      token,
+      userType,
+    }: {
+      euuid: ChildAccountPayload['euuid'];
+      token: string;
+      userType: Omit<UserType, 'child'>;
+    }) => {
+      try {
+        return await createChildAccountPersonalAccessToken({
+          euuid,
+          headers:
+            userType === 'proxy'
+              ? {
+                  Authorization: `Bearer ${token}`,
+                }
+              : undefined,
+        });
+      } catch (error) {
+        setIsProxyTokenError(error as APIError[]);
+        throw error;
+      }
+    },
+    []
+  );
+
+  // Navigate to the current location, triggering a re-render without a full page reload.
+  const refreshPage = React.useCallback(() => {
+    // TODO: Parent/Child: We need to test this against the real API.
+    history.push(history.location.pathname);
+  }, [history]);
+
+  const handleSwitchToChildAccount = React.useCallback(
+    async ({
+      currentTokenWithBearer,
+      euuid,
+      event,
+      handleClose,
+      isProxyUser,
+    }: {
+      currentTokenWithBearer?: AuthState['token'];
+      euuid: string;
+      event: React.MouseEvent<HTMLElement>;
+      handleClose: (e: React.SyntheticEvent<HTMLElement>) => void;
+      isProxyUser: boolean;
+    }) => {
+      try {
+        // TODO: Parent/Child: FOR MSW ONLY, REMOVE WHEN API IS READY
+        // ================================================================
+        // throw new Error(
+        //   `Account switching failed. Try again.`
+        // );
+        // ================================================================
+
+        // We don't need to worry about this if we're a proxy user.
+        if (!isProxyUser) {
+          const parentToken = {
+            expiry: storage.authentication.expire.get(),
+            scopes: storage.authentication.scopes.get(),
```

Comment:
```suggestion
            expiry: getStorage('authenication/expire'),
            scopes: getStorage('authenication/scopes'),
```
Would prevent us from having to import `storage` in this file, since we're already using `getStorage()` elsewhere.